tests/predict/test_predict.py

```diff
@@ -684,3 +686,42 @@ def test_predicted_outputs_piped_from_predict_to_lm_call():
         program(question="Why did a chicken cross the kitchen?", prediction="To get to the other side!")
 
     assert "prediction" not in mock_completion.call_args[1]
+
+
+def test_pydantic_special_type_serialization():
```

Comment:
the test name is bit vague, maybe something like `test_predict_dump_state_handles_non_json_types`?